### PR TITLE
Teach no-vars about array destructuring with ignored values

### PR DESCRIPTION
**Pull request by lencioni**
While running this on Airbnb code, I ran into the following error:

> TypeError: Cannot read property 'type' of null

Which seemed to happen when encountering array destructuring with
ignored values, like:

```js
var [, foo, bar] = baz;
```

I fixed this by guarding against this case in a few places.

While I was at it, I also fixed some typos that I happened to notice.

### CODE REVIEW

**Comment by cpojer** [comment]
I'm curious what kind of code you guys are writing. This pattern is so bad but I've needed it myself a couple of times. Thanks for the codemod fixes, as always!

**Comment by lencioni** [comment]
It looks like we only had that pattern in a couple of places. One where we were splitting a string and just wanted the second bit, and another where we were doing a regex match and we didn't care about the first element in the array.

**Comment by cpojer** [comment]
Nice, those seem like valid use cases. I didn't even realize one could skip variables in array destructuring, NICE.

**Comment by cpojer** [comment]
@lencioni oh, also, I sent you a message on Facebook if you'd be so kind to respond :)

File: transforms/no-vars.js [hunk 1]
```diff
@@ -38,7 +38,15 @@ export default function(file, api) {
...
+        e => {
+          if (!e) {
+            return null;
+          }
+          if (e.type === 'RestElement') {
+            return e.argument.name;
+          }
+          return e.name;
+        }
       );
```
Comment (ForbesLindesay) [inline transforms/no-vars.js]: You could add a `.filter(name => name !== null)` here to avoid the `null` checks elsewhere.

**Comment by ForbesLindesay** [comment]
Good catch :+1:

**Pull request merged**